### PR TITLE
feat(chapter_backtracking): Add js and ts codes for chapter 13.3

### PR DESCRIPTION
**Pull request by yuan0221**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: codes/typescript/chapter_divide_and_conquer/build_tree.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        hmap.set(inorder[i], i);
+    }
+    const root = dfs(
+        preorder,
+        inorder,
+        hmap,
+        0,
+        0,
+        inorder.length - 1
+    ) as TreeNode;
```
[the remark this answers is not in the document]
Comment (justin-tse) [reply codes/typescript/chapter_divide_and_conquer/build_tree.ts:47]: I don't think we need to use as here.

**Review by justin-tse — COMMENTED** [review]
@yuan0221 Thank you for your clean code! Please address the following code.

**Comment by yuan0221** [comment]
all done，thx.

**Review by justin-tse — COMMENTED** [review]
Great! Thank you! @yuan0221

**Pull request merged**